### PR TITLE
Enable nullability in DesignerToolStripControlHost

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in DesignerToolStripControlHost.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11211)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerToolStripControlHost.cs [hunk 1]
```diff
@@ -32,7 +31,20 @@ protected override Size DefaultSize
...
-        _behaviorService ??= (BehaviorService)parent.Site.GetService(typeof(BehaviorService));
+        if (_behaviorService is null)
+        {
+            if (!parent.Site.TryGetService(out BehaviorService? behaviorService))
+            {
+                throw new InvalidOperationException();
+            }
+
+            _behaviorService = behaviorService;
+        }
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerToolStripControlHost.cs:42]: I believe we have `GetRequiredService` available here
```suggestion
        _behaviorService ??= parent.Site.GetRequiredService<BehaviorService>();
```

**Review by lonitra — COMMENTED** [review]
Is it possible to add some test coverage here?

**Comment by gpetrou** [comment]
> Is it possible to add some test coverage here?

Hmm, how is GetGlyphs being used? I don't see any usages.

**Comment by Tanya-Solyanik** [comment]
Ultimately this method is used when a toolstrip on the design surface gets selected to draw the overlay glyphs on top of toolstrip items. Maybe we hadn't ported the relevant designers completely. ToolStrip designer is referenced by multiple other components and initially was ported only to support other designers. Even if this method is not used right now, I would keep it in case we enable the full scenario eventually.

**Comment by lonitra** [comment]
If we hadn't ported relevant designers completely we can wait until then to add tests. @gpetrou please address feedback above about `GetRequiredService` and we can get this to completion.

**Review by lonitra — APPROVED** [review]
Thank you

**Pull request merged**